radar_control/radar_control.py

```diff
@@ -596,25 +625,75 @@ def radar():
+                            # be ready to go if the first scan aligns directly to the minute. The
+                            # rest will need to wait until their boundary time is up.
+                            beam_scanbound = start_scan + timedelta(seconds=scan.scanbound[beam_iter])
+                            time_diff = beam_scanbound - datetime.utcnow()
+                            if time_diff.total_seconds() > 0:
+                                msg = "{}s until beam {} at time {}"
+                                msg = msg.format(sm.COLOR("blue", time_diff.total_seconds()),
+                                                sm.COLOR("yellow", beam_iter),
+                                                sm.COLOR("red", beam_scanbound))
+                                rad_ctrl_print(msg)
+                                time.sleep(time_diff.total_seconds())
+                            else:
+                                msg = "starting beam {} at time {}"
+                                msg = msg.format(sm.COLOR("yellow", beam_iter),
+                                                 sm.COLOR("red", beam_scanbound))
+                                rad_ctrl_print(msg)
+
+                            integration_period_start_time = datetime.utcnow()  # ms
+                            msg = "Integration start time: {}"
+                            msg = msg.format(sm.COLOR("red", integration_period_start_time))
+                            rad_ctrl_print(msg)
+
+                            # Here we find how much system time has elapsed to find the true amount
+                            # of time we can integrate for this scan boundary. We can then see if
+                            # we have enough time left to run the integration period.
+                            time_elapsed = integration_period_start_time - start_scan
+                            if beam_iter < len(scan.scanbound) - 1:
+                                scanbound_time = scan.scanbound[beam_iter+1] - scan.scanbound[0]
+                                beam_time = scanbound_time - time_elapsed.total_seconds()
+                            else:
+                                bound_minute = round_up_time(integration_period_start_time +
+                                                timedelta(milliseconds=aveperiod.intt))
+                                beam_time = bound_minute - integration_period_start_time
+                                beam_time = beam_time.total_seconds()
+
+                            msg = "beam {}: beam_time {}s"
+                            msg = msg.format(sm.COLOR("yellow", beam_iter),
+                                             sm.COLOR("blue", beam_time))
+                            rad_ctrl_print(msg)
+
+                            if beam_time < aveperiod.intt*1e-3:
+                                    #TODO maybe log skipping beam?
+                                    msg = "Scan time too long, skipping beam {}"
```

Comment:
as we discussed, perhaps we can just shorten up the number of averages for this beam if it's late to start, rather than skipping it altogether.

Reply:
Yeah lets look at this later. With adequate intt, we dont even need to align to the beam since our timing is much better than qnx.